pkg/istio/virtualservice.go

```diff
@@ -97,16 +97,19 @@ func mutateVirtualService(ctx model.SessionContext, sourceResource model.Located
 		routes := []*v1alpha3.HTTPRoute{}
 		for _, h := range vs.Spec.Http {
 			for _, r := range h.Route {
-				if r.Destination.Host == host && r.Destination.Subset == subset {
-					routes = append(routes, h)
+				if r.Destination != nil && r.Destination.Host == host {
+					if r.Destination.Subset == "" || r.Destination.Subset == subset {
+						routes = append(routes, h)
+					}
 				}
 			}
 		}
 		return routes
 	}
 	removeOtherRoutes := func(http v1alpha3.HTTPRoute, host, subset string) v1alpha3.HTTPRoute {
 		for i, r := range http.Route {
-			if !(r.Destination.Host == host && r.Destination.Subset == subset) {
+			if !((r.Destination.Host == host && r.Destination.Subset == subset) ||
```

Comment:
Could `r.Destination` be `nil` like in L100?

Reply:
Technically it shouldn't be nil at this point. That's more of a copy paste error. The mutationRequired should block that. (if check to even enter this part). As well as Destination being required on the API level.

Reply:
https://github.com/Maistra/istio-workspace/pull/308/commits/7f3de5bb207a9d796f96b2cdd97d307b0d1dcc4f